WordPress/gutenberg
# ToolsPanel component: refactor  to typescript

## Description
Converts the ToolsPanel component to typescript. Also adds type annotations to the Menu components that it uses.

## How has this been tested?

- Check that linting and typescript compile checks all complete successfully
- Add margin support to the Group block and add a group block and make sure the dimensions panel all works as expected.
